ionoscloud/resource_server_cube.go

```diff
@@ -0,0 +1,1075 @@
+	}
+
+	if v, ok := d.GetOk("volume.0.availability_zone"); ok {
+		vStr := v.(string)
+		volume.AvailabilityZone = &vStr
+	}
+
+	if v, ok := d.GetOk("volume.0.name"); ok {
+		vStr := v.(string)
+		volume.Name = &vStr
+	}
+
+	if v, ok := d.GetOk("volume.0.bus"); ok {
+		vStr := v.(string)
+		volume.Bus = &vStr
+	}
+
+	if v, ok := d.GetOk("volume.0.backup_unit_id"); ok {
+		vStr := v.(string)
+		volume.BackupunitId = &vStr
+	}
+
+	if v, ok := d.GetOk("volume.0.user_data"); ok {
+		vStr := v.(string)
+		volume.UserData = &vStr
+	}
+
+	if v, ok := d.GetOk("volume.0.ssh_key_path"); ok {
+		sshKeyPath = v.([]interface{})
+		if err := d.Set("ssh_key_path", v.([]interface{})); err != nil {
+			diags := diag.FromErr(err)
+			return diags
+		}
+	} else if v, ok := d.GetOk("ssh_key_path"); ok {
+		sshKeyPath = v.([]interface{})
+		if err := d.Set("ssh_key_path", v.([]interface{})); err != nil {
+			diags := diag.FromErr(err)
+			return diags
+		}
+	} else {
+		if err := d.Set("ssh_key_path", [][]string{}); err != nil {
+			diags := diag.FromErr(err)
+			return diags
+		}
+	}
+
+	if _, ok := d.GetOk("boot_cdrom"); ok {
+		resId := d.Get("boot_cdrom").(string)
+		server.Properties.BootCdrom = &ionoscloud.ResourceReference{
+			Id: &resId,
+		}
+	}
+
+	if _, ok := d.GetOk("boot_volume"); ok {
+		diags := diag.FromErr(fmt.Errorf("boot_volume argument can be set only in update requests \n"))
+		return diags
+	}
+
+	if len(sshKeyPath) != 0 {
+		for _, path := range sshKeyPath {
+			log.Printf("[DEBUG] Reading file %s", path)
+			publicKey, err := readPublicKey(path.(string))
+			if err != nil {
+				diags := diag.FromErr(fmt.Errorf("error fetching sshkey from file (%s) %w", path, err))
+				return diags
+			}
+			publicKeys = append(publicKeys, publicKey)
+		}
+		if len(publicKeys) > 0 {
+			volume.SshKeys = &publicKeys
+		}
+	}
+
+	if v, ok := d.GetOk("image_name"); ok {
+		imageInput = v.(string)
+	}
+
+	if imageInput != "" {
+		image, imageAlias, isSnapshot, diags = checkImage(ctx, client, imageInput, password, licenceType, dcId, sshKeyPath)
+		if diags != nil {
+			return diags
+		}
+	}
+
+	if isSnapshot == true && (volume.ImagePassword != nil && *volume.ImagePassword != "" || len(publicKeys) > 0) {
+		diags := diag.FromErr(fmt.Errorf("you can't pass 'image_password' and/or 'ssh keys' when creating a volume from a snapshot"))
+		return diags
+	}
+
+	if image != "" {
+		volume.Image = &image
+	} else {
+		volume.Image = nil
+	}
+
+	if imageAlias != "" {
+		volume.ImageAlias = &imageAlias
+	} else {
+		volume.ImageAlias = nil
+	}
+
+	server.Entities = &ionoscloud.ServerEntities{
+		Volumes: &ionoscloud.AttachedVolumes{
+			Items: &[]ionoscloud.Volume{
+				{
+					Properties: &volume,
+				},
+			},
+		},
+	}
+	var primaryNic *ionoscloud.Nic
+	if server.Entities.Nics != nil && server.Entities.Nics.Items != nil && len(*server.Entities.Nics.Items) > 0 {
+		primaryNic = &(*server.Entities.Nics.Items)[0]
+	}
+	// Nic Arguments
+	if _, ok := d.GetOk("nic"); ok {
```

Comment:
should be the same for server I think

Reply:
I think i can use the func from server.go